### DISCUSSION TITLE
Json files under config/themes/[theme] should be deleted when cache is emptied

### DISCUSSION THREAD

**Discussion by web-cooking-factory** [PrestaShop/PrestaShop#31719] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

If we edit the theme.yml of our themes, we have to manually delete the json files generated under config/themes/[theme] in order to make the modifications effective.

### Expected behavior

Update the json files when we empty the cache in the BO

### Steps to reproduce

1. Edit theme.yml of classic theme, by changing layout on category page from layout-left-column to layout-full-width for instance
2. Check in FO, the category page is still displayed in layout left column (if there are modules hooked on displayLeftColumn)
3. Manually delete the json files
4. Check the FO, the category page now has the layout full width

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

7.4

**Comment by Hlavtox** [comment]
@web-cooking-factory I must disagree. The problem is that by deleting the JSON you would loose all layout customizations set in `Backoffice > Theme & Logo > Choose layouts`, that's not what we want. Usually, you need to delete that JSON only if you changed for example a theme name and you don't want to update it through BO.

Ping @kpodemski

**Comment by web-cooking-factory** [comment]
Not only, this can happen when you edit assets as well for instance.
There should be a way to make the modification effective without directly deleting the files.

**Comment by kpodemski** [comment]
@web-cooking-factory 

I agree. We already have a "Reset theme" button (in dev mode), maybe we could have a "Refresh theme", but we would need to be explicit on what it does.

One more thing. I wouldn't consider this a bug because, as @Hlavtox noticed, the consequences would be harmful if we refresh the theme every time somebody clears the cache.

**Comment by web-cooking-factory** [comment]
What would be convenient is the choice of what we want to reset, mainly assets, hooks, layouts.
As front-end dev, I often edit assets, sometimes layouts, hooks never.

**Comment by web-cooking-factory** [comment]
> One more thing. I wouldn't consider this a bug because, as @Hlavtox noticed, the consequences would be harmful if we refresh the theme every time somebody clears the cache.

Improvement ?

**Comment by kpodemski** [comment]
@web-cooking-factory yeah, this would be a new feature in reality, I understand the need, and I'll transfer this issue to "Discussions", maybe we will come up with some valid solution. Thanks for raising this subject 👍🏻

**Comment by PrestaEdit** [comment]
This is something that I need to work in internal as it's something needed for me : add an asset without reset the theme.

I make my vote on this.
